common/optimiser/Optimiser.java

```diff
@@ -45,30 +53,49 @@ public class Optimiser {
     private MPSolver solver;
     private MPSolverParameters parameters;
     private Status status;
-    private boolean hasSolver;
-    private double objectiveValue;
+    private boolean isConstraintsUpToDate;
+    private Double objectiveValue;
 
     public Optimiser() {
         variables = new ArrayList<>();
         constraints = new HashSet<>();
         objectiveCoefficients = new HashMap<>();
         status = Status.NOT_SOLVED;
-        hasSolver = false;
+        objectiveValue = null;
+        isConstraintsUpToDate = false;
     }
 
     public synchronized Status optimise(long timeLimitMillis) {
-        assert hasSolver;
         if (isOptimal()) return status;
+        assert solutionSatisfiesModel();
+        maySetSolver();
+        setSolutionAsHints();
```

Comment:
always apply the previous solution as hints to the model - since the solver always restarts this is the only way to "carry on" solving from the last solution